pytorch_lightning/core/lightning.py

```diff
@@ -1586,6 +1594,7 @@ def optimizer_step(
         optimizer_idx: int = 0,
         optimizer_closure: Optional[Callable[[], Any]] = None,
         on_tpu: bool = False,
+        on_hpu: bool = None,
```

Comment:
This is an API BC. Let's not add it there and I believe it is not necessary either. the LightningOptimizer should handle the optimizer.step properly.
We should depreciate on_tpu going forward as it can be accessed from the LightningModule property.